lupa/_lupa.pyx

```diff
@@ -1916,3 +1917,35 @@ cdef void luaL_openlib(lua_State *L, const char *libname,
         luaL_setfuncs(L, l, nup)
     else:
         lua.lua_pop(L, nup)
+
+
+# protected calls for manipulating the Lua stack in Python
+
+cdef int luaP_gettable_aux(lua_State* L) nogil:
+    lua.lua_pushvalue(L, lua.lua_upvalueindex(1))  # key
+    lua.lua_gettable(L, lua.lua_upvalueindex(2))   # tbl[key]
+    return 1
+
+
+cdef luaP_gettable(LuaRuntime runtime, lua_State* L, int t):
```

Comment:
Please don't use `lua…` as prefix for module internal functions. That's confusing because they look like Lua API functions. `_get_from_lua_table` seems an appropriate name. Also, please give `t` a name that suggests a reference to a stack index, e.g. `table_ref` or `table_on_stack` or something better that I'm sure you can come up with.

Actually, why is `t` needed at all? Can't we just let the caller make sure that the stack contains [table, key]` in the right order? After all, this is only used in a single place (for now).

Reply:
Agreed!